### ISSUE TITLE
Support incremental builds

### ISSUE THREAD

**Issue by tompazourek** [ym-project/gulp-esbuild#1]
esbuild v0.8.12 added support for incremental builds via the `incremental: true` option.

What's your opinion on adding support for this to `gulp-esbuild`? I think it would be nice to have.

**Comment by ym-project** [comment]
Hi @tompazourek!
I saw this feature. This feature would be suit to gulp watch mode perfectly.

I made a new branch and an [example](https://github.com/ym-project/gulp-esbuild/tree/watch-mode/examples/watch). You can check it (instructions in README).

It works such way:
```js
const {createGulpEsbuild} = require('gulp-esbuild')
const gulpEsbuild = createGulpEsbuild()

function gulpTask() {
    return src()
        .pipe(gulpEsbuild())
        .pipe(dest())
}

watch('', gulpTask)
```
Wrapper `createGulpEsbuild` needs to remember is is the first build or more than first build.

If you like it I will release new version. If you have a notice I will be glad to read it.

**Comment by ym-project** [comment]
I tested it and got results:
```sh
[14:38:42] Using gulpfile ~.../gulp-esbuild/examples/watch/gulpfile.js
[14:38:42] Starting 'watch'...
[14:38:50] Starting 'build'...
[14:38:50] Finished 'build' after 156 ms
[14:38:54] Starting 'build'...
[14:38:54] Finished 'build' after 60 ms
[14:39:17] Starting 'build'...
[14:39:17] Finished 'build' after 62 ms
```
Rebuild is faster 2.5x than the first build.

**Comment by tompazourek** [comment]
@ym-project Looks really good, thank you for sharing! 👍 I'm just experimenting at this point, I'll have to try this out.

**Comment by ym-project** [comment]
I released a new version with incremental build (0.3.0 version). I think this issue is resolved. If you will find any problems, just create a new issue.
